### ISSUE TITLE
pgbackrest 2.19 don't work with postgres datadir symlinked

### ISSUE THREAD

**Issue by Yuxael** [pgbackrest/pgbackrest#882]
1. pgBackRest version:
2.19

2. PostgreSQL version:
11

3. Operating system/version:
Ubuntu 16.04

4. Did you install pgBackRest from source or from a package?
Package

5. Log:
```
2019-11-29 18:34:43.099 P00 ASSERT: [025]: absolute path '/srv/var/lib/postgresql/11/main/pg_wal/archive_status' is not in base path '/var/lib/postgresql/11/main'
                                    STACK TRACE:
                                    storage/storage:storagePath:674:(test build required for parameters)
                                        ... function(s) omitted ...
                                    storage/storage:storageList:(trace log level required for parameters)
                                    command/archive/push/push:archivePushReadyList:(trace log level required for parameters)
                                    command/archive/push/push:archivePushProcessList:(trace log level required for parameters)
                                    command/archive/push/push:cmdArchivePushAsync:(debug log level required for parameters)
                                    main:main:(debug log level required for parameters)
2019-11-29 18:34:43.099 P00   INFO: archive-push-async command end: aborted with exception [025]
```
7. Describe the issue:
After updating pgbackrest to 2.19 async push stopped working because whole /var directory is symlinked to /srv/var.

I think this is quite bad, since 2.18 and previous versions worked well with symlinks and now it's broken. There should be at least info in changelog.

Also when pg1-path is set to actual path, not symlink I get this:
```
ERROR: [058]: version '11' and path '/var/lib/postgresql/11/main' queried from cluster do not match version '11' and '/srv/var/lib/postgresql/11/main' read from '/srv/var/lib/postgresql/11/main/global/pg_control'
```

**Comment by dwsteele** [comment]
This is a bug introduced in 2.19 in this commit: 7168e074401.  pgBackRest is now using cwd() to construct the absolute path when a relative path is passed.  But, the storage layer also tries to validate that it is reading from the right place and it is now getting an unexpected path.

We'll fix this in the next release, but for now you should be able to pass `/var/lib/postgresql/11/main/%p` in your archive command to fix it.

You can't change `pg1-path` in pgBackRest without changing `data_directory` in Postgres since they get compared before backup.

**Comment by dwsteele** [comment]
This has been fixed in 0194a9867182435c7a which will go out with the 2.20 release this week.
